src/main/java/net/jodah/failsafe/ExecutionContext.java

```diff
@@ -44,6 +49,13 @@
   public Duration getElapsedTime() {
     return new Duration(System.nanoTime() - startTime.toNanos(), TimeUnit.NANOSECONDS);
   }
+  
+  /**
+   * Returns the elapsed time since current execution began. If it never started, this will be null.
+   */
+  public Duration getElapsedAttemptTime() {
```

Comment:
We can't use Optional here since Failsafe is Java 6 compatible... this would throw a ClassNotFoundException (or something similar) for pre-Java 8 users.

Reply:
changed it